msu/hooks/skills/skill.nut

```diff
@@ -0,0 +1,1052 @@
+				this.addDamageType(::Const.Damage.DamageType.Piercing);
+				break;
+		}
+	}
+
+	o.verifyTargetAndRange <- function( _targetTile, _origin = null )
+	{
+		if (_origin == null)
+		{
+			_origin = this.getContainer().getActor().getTile();	
+		}
+		
+		return this.onVerifyTarget(_origin, _targetTile) && this.isInRange(_targetTile, _origin);
+	}
+
+	local getDescription = o.getDescription;
+	o.getDescription = function()
+	{
+		if (this.m.DamageType.len() == 0)
+		{
+			return getDescription();
+		}
+
+		local ret = "[color=" + ::Const.UI.Color.NegativeValue + "]Inflicts ";
+
+		foreach (d in this.m.DamageType)
+		{
+			local probability = ::Math.round(this.getDamageTypeProbability(d.Type) * 100);
+
+			if (probability < 100)
+			{
+				ret += probability + "% ";
+			}
+			
+			ret += ::Const.Damage.getDamageTypeName(d.Type) + ", ";
+		}
+
+		ret = ret.slice(0, -2);
+
+		ret += " Damage [/color]\n\n" + getDescription();
+
+		return ret;
+	}
+
+	local getHitFactors = o.getHitFactors;
+	o.getHitFactors = function( _targetTile )
+	{
+		local ret = getHitFactors(_targetTile);			
+		this.getContainer().onGetHitFactors(this, _targetTile, ret);
+		return ret;
+	}
+
+	local getDefaultTooltip = o.getDefaultTooltip;
+	o.getDefaultTooltip = function()
+	{
+		local tooltip = getDefaultTooltip();
+
+		if (this.isRanged())
+		{
+			local rangeBonus = ", more";
+			if (this.m.MaxRangeBonus == 0)
+			{
+				rangeBonus = " or"
+			}
+			else if (this.m.MaxRangeBonus < 0)
+			{
+				rangeBonus = ", less"
+			}
+
+			tooltip.push({
+				id = 6,
+				type = "text",
+				icon = "ui/icons/vision.png",
+				text = "Has a range of [color=" + ::Const.UI.Color.PositiveValue + "]" + this.getMaxRange() + "[/color] tiles on even ground" + rangeBonus + " if shooting downhill"
+			});
+
+			local accuText = "";
+			if (this.m.AdditionalAccuracy != 0)
+			{
+				local color = this.m.AdditionalAccuracy > 0 ? ::Const.UI.Color.PositiveValue : ::Const.UI.Color.NegativeValue;
+				local sign = this.m.AdditionalAccuracy > 0 ? "+" : "";
+				accuText = "Has [color=" + color + "]" + sign + this.m.AdditionalAccuracy + "%[/color] chance to hit";
+			}
+
+			if (this.m.AdditionalHitChance != 0)
+			{
+				accuText += this.m.AdditionalAccuracy == 0 ? "Has" : ", and";
+				local additionalHitChance = this.m.AdditionalHitChance + this.getContainer().getActor().getCurrentProperties().HitChanceAdditionalWithEachTile;
+				local sign = additionalHitChance > 0 ? "+" : "";
+				accuText += " [color=" + (additionalHitChance > 0 ? ::Const.UI.Color.PositiveValue : ::Const.UI.Color.NegativeValue) + "]" + sign + additionalHitChance + "%[/color]";
+				accuText += this.m.AdditionalAccuracy == 0 ? " chance to hit " : "";
+				accuText += " per tile of distance";
+			}
+
+			if (accuText.len() != 0)
+			{
+				tooltip.push({
+					id = 7,
+					type = "text",
+					icon = "ui/icons/hitchance.png",
+					text = accuText
+				});
+			}
+		}
+
+		return tooltip;
+	}
```

Comment:
This is an issue. Because in vanilla it will end up duplicating the info and showing it twice in the tooltip as vanilla tooltips of ranged skills e.g. `aimed_shot` first call `getDefaultTooltip` and then add this info manually. Therefore, perhaps the best solution is to go back to being a `getDefaultRangedTooltip` function that modders can use. This way we leave vanilla skills alone.